disttask/framework/storage/task_table.go

```diff
@@ -510,6 +510,54 @@ func (stm *TaskManager) GetSchedulerIDsByTaskID(taskID int64) ([]string, error)
 	return instanceIDs, nil
 }
 
+// IsSchedulerCanceled checks if subtask 'execID' of task 'taskID' has been canceled somehow.
+func (stm *TaskManager) IsSchedulerCanceled(taskID int64, execID string) (bool, error) {
+	rs, err := stm.executeSQLWithNewSession(stm.ctx, "select distinct(exec_id) from mysql.tidb_background_subtask where task_key = %? and exec_id = %?", taskID, execID)
```

Comment:
```suggestion
	rs, err := stm.executeSQLWithNewSession(stm.ctx, "select 1 from mysql.tidb_background_subtask where task_key = %? and exec_id = %?", taskID, execID)
```

Use `count(1)` or `1`?